### PR TITLE
fix grammatical issues in how to use promises

### PR DESCRIPTION
**Pull request by yashikabhargava**
### Description

fixed minor grammatical issues in how to use promises under introduction to asynchronous javascript.

### CODE REVIEW

File: files/en-us/learn/javascript/asynchronous/promises/index.md [hunk 1]
```diff
@@ -168,7 +168,7 @@ Promises come with some quite specific terminology that it's worth getting clear
 First, a promise can be in one of three states:
 
 - **pending**: the promise has been created, and the asynchronous function it's associated with has not succeeded or failed yet. This is the state your promise is in when it's returned from a call to `fetch()`, and the request is still being made.
-- **fulfilled**: the asynchronous function has succeeded. When a promise is fulfilled, its `then()` handler is called.
+- **fulfilled**: the asynchronous function has succeeded. When a promise is fulfilled, it's `then()` handler is called.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/asynchronous/promises/index.md:171]: Original is correct

File: files/en-us/learn/javascript/asynchronous/promises/index.md [hunk 2]
```diff
@@ -183,12 +183,12 @@ The article [Let's talk about how to talk about promises](https://thenewtoys.dev
 
 The promise chain is what you need when your operation consists of several asynchronous functions, and you need each one to complete before starting the next one. But there are other ways you might need to combine asynchronous function calls, and the `Promise` API provides some helpers for them.
 
-Sometimes, you need all the promises to be fulfilled, but they don't depend on each other. In a case like that, it's much more efficient to start them all off together, then be notified when they have all fulfilled. The {{jsxref("Promise/all", "Promise.all()")}} method is what you need here. It takes an array of promises and returns a single promise.
+Sometimes, you need all the promises to be fulfilled, but they don't depend on each other. In a case like that, it's much more efficient to start them all off together, then be notified when they have all been fulfilled. The {{jsxref("Promise/all", "Promise.all()")}} method is what you need here. It takes an array of promises and returns a single promise.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/asynchronous/promises/index.md:186]: Original is correct; we can say "a promise fulfills" or "a promise rejects"

File: files/en-us/learn/javascript/asynchronous/promises/index.md [hunk 3]
```diff
@@ -183,12 +183,12 @@ The article [Let's talk about how to talk about promises](https://thenewtoys.dev
...
 
 The promise returned by `Promise.all()` is:
 
 - fulfilled when and if _all_ the promises in the array are fulfilled. In this case, the `then()` handler is called with an array of all the responses, in the same order that the promises were passed into `all()`.
-- rejected when and if _any_ of the promises in the array are rejected. In this case, the `catch()` handler is called with the error thrown by the promise that rejected.
+- rejected when and if _any_ of the promises in the array are rejected. In this case, the `catch()` handler is called with the error thrown by the promise that was rejected.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/asynchronous/promises/index.md:191]: Same here

File: files/en-us/learn/javascript/asynchronous/promises/index.md [hunk 4]
```diff
@@ -258,7 +258,7 @@ Promise.any([fetchPromise1, fetchPromise2, fetchPromise3])
   });
 ```
 
-Note that in this case we can't predict which fetch request will complete first.
+Note that in this case we can't predict which fetch request will be completed first.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/asynchronous/promises/index.md:261]: Mixed feelings about this one; I don't think the original one is wrong

**Review by Josh-Cena — COMMENTED** [review]
A lot of things which I feel that don't need to be changed

File: files/en-us/learn/javascript/asynchronous/promises/index.md (hunk 4 above)
Comment (yashikabhargava) [reply files/en-us/learn/javascript/asynchronous/promises/index.md:261]: okay, replacing all with the original one.

File: files/en-us/learn/javascript/asynchronous/promises/index.md [hunk 5]
```diff
@@ -373,7 +373,7 @@ The `async` and `await` keywords make it easier to build an operation from a ser
 
 Promises work in the latest versions of all modern browsers; the only place where promise support will be a problem is in Opera Mini and IE11 and earlier versions.
 
-We didn't touch on all promise features in this article, just the most interesting and useful ones. As you start to learn more about promises, you'll come across more features and techniques.
+We didn't touch on all the features of promises in this article, just the most interesting and useful ones. As you start to learn more about promises, you'll come across more features and techniques.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/asynchronous/promises/index.md:376]:
```suggestion
We didn't touch on all features of promises in this article, just the most interesting and useful ones. As you start to learn more about promises, you'll come across more features and techniques.
```

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**